trait2d/analysis/__init__.py

```diff
@@ -471,23 +473,65 @@ def adc_summary(self, avg_only_params = False, interpolation = False, plot_msd =
         for model in counter:
             sector[model] = counter[model] / len(self._tracks)
         sector["not catergorized"] = (len(self._tracks) - counter_sum) / len(self._tracks)
+        
+        ###ENSEMBLE AVERAGE - Added by FR###
+        
+        ens_D_app = np.zeros(track_length - 3)
+        ens_MSD = np.zeros(track_length - 3)
+        
+        if ensemble_average:
+            for track in self._tracks:
```

Comment:
I would move the definitions inside the if statement (they're not needed outside). Also, see below for the `sampled` variable.

```suggestion
        if ensemble_average:
            ens_D_app = np.zeros(track_length - 3)
            ens_MSD = np.zeros(track_length - 3)
            sampled = np.zeros(track_length - 3)
            for track in self._tracks:
```

Reply:
I have implemented the changes, with some modifications to avoid some confusion (mostly on my part). Should work fine now.